provider/resource_kpi_threshold_template.go

```diff
@@ -39,209 +44,431 @@ func kpiThresholdTemplateTFFormat(b *models.Base) (string, error) {
-				Type:        schema.TypeString,
+			"sec_grp": schema.StringAttribute{
 				Required:    true,
 				Description: "The team the object belongs to. ",
 			},
-			"time_variate_thresholds_specification": {
-				Required:    true,
-				Type:        schema.TypeSet,
-				Description: "Data structure for time variate threshold specification.",
-				Elem: &schema.Resource{
-					Schema: getKpiThresholdPolicySchema(),
-				},
-			},
 		},
 	}
 }
 
-func kpiThresholdTemplate(ctx context.Context, d *schema.ResourceData, clientConfig models.ClientConfig) (config *models.Base, err error) {
-	body := map[string]interface{}{}
-	body["objectType"] = "kpi_threshold_template"
-	body["title"] = d.Get("title").(string)
-	body["description"] = d.Get("description").(string)
-	body["adaptive_thresholds_is_enabled"] = d.Get("adaptive_thresholds_is_enabled").(bool)
-	body["adaptive_thresholding_training_window"] = d.Get("adaptive_thresholding_training_window").(string)
-	body["time_variate_thresholds"] = d.Get("time_variate_thresholds").(bool)
-	body["sec_grp"] = d.Get("sec_grp").(string)
-
-	timeVariateThresholdsSpecification := map[string]interface{}{}
-	for _, sourceTimeVariateSpecification_ := range d.Get("time_variate_thresholds_specification").(*schema.Set).List() {
-		sourceTimeVariateSpecification := sourceTimeVariateSpecification_.(map[string]interface{})
-		policies := map[string]interface{}{}
-		for _, sourcePolicy_ := range sourceTimeVariateSpecification["policies"].(*schema.Set).List() {
-			sourcePolicy := sourcePolicy_.(map[string]interface{})
-			policyName := sourcePolicy["policy_name"].(string)
-
-			policies[policyName], err = kpiThresholdPolicyToPayload(sourcePolicy)
-			if err != nil {
-				return nil, err
-			}
-		}
-		timeVariateThresholdsSpecification["policies"] = policies
+func (r *resourceKpiThresholdTemplate) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state modelKpiThresholdTemplate
+	diags := req.Plan.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	body["time_variate_thresholds_specification"] = timeVariateThresholdsSpecification
 
-	base := kpiThresholdTemplateBase(clientConfig, d.Id(), d.Get("title").(string))
-	err = base.PopulateRawJSON(ctx, body)
-
-	return base, err
-}
-
-func kpiThresholdTemplateCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
-	template, err := kpiThresholdTemplate(ctx, d, m.(models.ClientConfig))
+	template, err := kpiThresholdTemplate(ctx, state, r.client)
 	if err != nil {
-		return diag.FromErr(err)
+		diags.AddError("Failed to populate kpi threshold template.", err.Error())
+		resp.Diagnostics.Append(diags...)
+		return
 	}
+
 	b, err := template.Create(ctx)
 	if err != nil {
-		return diag.FromErr(err)
+		diags.AddError("Failed to create kpi threshold template.", err.Error())
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	_, err = b.Read(ctx)
```

Comment:
Probably don't need to read it back. Hopefully we can just rely on the ITSI API to give us an error if it actually fails to create it. And it looks like in the old resource, we weren't using the `Read` results as well.